src/provider/controller/NotificationCenter.ts

```diff
@@ -1,145 +1,186 @@
-
-
-    private setupTrayIcon() {
-        const application = fin.Application.getCurrentSync();
-        const icon = 'https://openfin.co/favicon-32x32.png';
-
-        application.addListener('tray-icon-clicked', (event: TrayIconClicked<string, string>) => {
-            console.log(event);
-            if (event.button === 0) {
-                this.toggleWindow();
-            }
-        });
-
-        application.setTrayIcon(icon);
+    private async subscribe(): Promise<void> {
+        // Window visibility
+        this._store.watchForChange(
+            state => state.windowVisible,
+            (_, value) => this.toggleWindow(value)
+        );
     }
 
     /**
-     * @function onWindowLoad Fired when the window DOM is loaded
-     * @returns void
+     * The window visibility state.
      */
-    public onWindowLoad(): void {
-        this.sizeToFit(false);
-
-        document.getElementById('exitLink')!.addEventListener('click', () => this.hideWindow());
+    public get visible(): boolean {
+        const state = this._store.state;
+        return state.windowVisible;
     }
 
     /**
-     * @function sizeToFit Sets the window dimensions in shape of a side bar
-     * @returns void
+     * Add listeners to the window.
      */
-    public sizeToFit(forceShow: boolean = false): void {
-        fin.desktop.System.getMonitorInfo((monitorInfo: fin.MonitorInfo) => {
-            this.windowInfo.getWindow().setBounds(
-                monitorInfo.primaryMonitor.availableRect.right - this.windowInfo.getIdealWidth(),
-                0,
-                this.windowInfo.getIdealWidth(),
-                monitorInfo.primaryMonitor.availableRect.bottom,
-                () => {
-                    if (forceShow) {
-                        this.showWindow();
-                    }
-                },
-                (reason: string) => {
-                    console.warn('MOVED FAILED', reason);
+    private async addListeners(): Promise<void> {
+        const {window} = this._webWindow;
+        const hideOnBlur = process.env.NODE_ENV === 'production';
+
+        if (hideOnBlur) {
+            window.addListener('blurred', async () => {
+                if (this.visible) {
+                    this._store.dispatch({type: Action.TOGGLE_VISIBILITY, visible: false});
                 }
-            );
-        });
+            });
+        }
+        fin.System.addListener('monitor-info-changed', ((event: MonitorEvent<string, string>) => {
+            this.sizeToFit();
+        }));
     }
 
     /**
-     * @function showWindow Shows the Window with Fade()
-     * @returns void
+     * Show the window.
      */
-    public showWindow(): void {
-        this.fade(true, 450);
-        this.windowInfo.getWindow().resizeBy(1, 0, 'top-left');
-        this.windowInfo.getWindow().resizeBy(-1, 0, 'top-left');
-        this.windowInfo.setShowing(true);
+    public async showWindow(): Promise<void> {
+        const {window} = await this._webWindow;
```

Comment:
Don't think the await is needed anymore? Here, or the other places in this file we have `await this._webWindow`. (Sorry)

Reply:
Nice catch, removed all the other instances of `await this._webWindow;`